Pinned jinja2 version is no longer supported and conflicts with the latest MarkupSafe

**Current behavior**

insights-core has pinned the jinja2 version to <=2.11.3 in #3057. This version is [no longer supported](https://github.com/pallets/jinja/issues/1585#issuecomment-1043892628).

**Expected behavior**

insights-core can be used with a maintained jinja2 version in applications that depend both on insights-core and jinja2.

**Impact**

jinja2-2.11.3 fails to import with the latest version of MarkupSafe, 2.1.0:

```
...
.tox/unit-py38/lib/python3.8/site-packages/jinja2/filters.py:13: in <module>
    from markupsafe import soft_unicode
E   ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/builds/ccx/ccx-rules-ocp/.tox/unit-py38/lib64/python3.8/site-packages/markupsafe/__init__.py
```

All projects that depend on insights-core have to pin unsupported versions of MarkupSafe now. I would argue that it is important to keep up-to-date for a package that is used for secure handling of user input.

--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,8 @@ runtime = set([
     'cachecontrol[filecache]',
     'defusedxml',
     'lockfile',
-    'jinja2<=2.11.3',
+    'jinja2<=2.11.3; python_version <= "2.7"',
+    'jinja2; python_version > "2.7"',
     'pyyaml>=3.10,<=3.13; python_version < "2.7"',
     'pyyaml; python_version >= "2.7"',
 ])
@@ -66,7 +67,6 @@ docs = set([
     'sphinx_rtd_theme',
     'ipython',
     'colorama',
-    'jinja2<=2.11.3',
     'Pygments',
     'jedi<0.18.0',    # Open issue with jedi 0.18.0 and iPython <= 7.19
                       # https://github.com/davidhalter/jedi/issues/1714

